packages/aws-appsync/src/link/complex-object-link.ts

```diff
@@ -83,39 +86,44 @@ const complexObjectFields = [
     { name: 'key', type: 'string' },
     { name: 'region', type: 'string' },
     { name: 'mimeType', type: 'string' },
-    { name: 'localUri', type: 'object' },
+    { name: 'localUri', type: ['object', 'string'] },
 ];
 const findInObject = obj => {
-    let which;
-
-    const _findInObject = obj => {
-        return Object.keys(obj).find(key => {
-            const val = obj[key];
+    const testFn = val => {
+        return complexObjectFields.every(field => {
+            const hasValue = val[field.name];
+            const types: string[] = Array.isArray(field.type) ? field.type : [field.type];
+            const isOfType = hasValue && types.reduce((prev, curr) => {
+                return prev || typeof val[field.name] === curr;
+            }, false);
+
+            return isOfType;
+        });
+    };
 
-            if (val && typeof val === 'object') {
-                const hasFields = complexObjectFields.every(field => {
-                    const hasValue = val[field.name];
-                    const types: string[] = Array.isArray(field.type) ? field.type : [field.type];
-                    const isOfType = hasValue && types.reduce((prev, curr) => {
-                        return prev || typeof val[field.name] === curr;
-                    }, false);
+    const _findInObject = (obj, path = '', acc = {}) => {
+        if (!obj) {
+            return acc;
+        }
 
-                    return isOfType;
-                });
+        if (testFn(obj)) {
+            acc[path] = { ...obj };
+            delete obj.mimeType;
```

Comment:
Why you need to delete those?

Reply:
Because once we upload to S3, there's no need to send the file contents and mimeType to AppSync too